packages/cli/src/commands/init/init.js

```diff
@@ -44,33 +89,28 @@ function adjustNameIfUrl(name, cwd) {
 async function createFromTemplate({
   projectName,
   templateName,
-  version,
   npm,
+  directory,
 }: {
   projectName: string,
   templateName: string,
-  version?: string,
   npm?: boolean,
+  directory: string,
 }) {
   logger.debug('Initializing new project');
   logger.log(banner);
+
+  await setProjectDirectory(directory);
+
   const Loader = getLoader();
   const loader = new Loader({text: 'Downloading template'});
   const templateSourceDir = fs.mkdtempSync(
     path.join(os.tmpdir(), 'rncli-init-template-'),
   );
 
-  if (version && semver.valid(version) && !semver.gte(version, '0.60.0-rc.0')) {
```

Comment:
Turned out we can now entirely extract version handling from `createFromTemplate`, which makes more sense because version is only a parameter of `templateName`. Thanks for making me think about it harder @grabbou :P

Reply:
No worries